server/athenian/api/experiments/preloading/entries.py

```diff
@@ -1,27 +1,93 @@
+    MetricEntriesCalculator as OriginalMetricEntriesCalculator
+from athenian.api.controllers.features.github.unfresh_pull_request_metrics import \
+    UnfreshPullRequestFactsFetcher as OriginalUnfreshPullRequestFactsFetcher
 from athenian.api.controllers.miners.filters import JIRAFilter, LabelFilter
-from athenian.api.controllers.miners.github.pull_request import (
-    PullRequestMiner as OriginalPullRequestMiner,
-)
+from athenian.api.controllers.miners.github.pull_request import \
+    PullRequestMiner as OriginalPullRequestMiner
+from athenian.api.controllers.miners.github.release_load import \
+    match_groups_to_conditions, ReleaseLoader as OriginalReleaseLoader
+from athenian.api.controllers.miners.github.release_load import \
+    remove_ambigous_precomputed_releases
+from athenian.api.controllers.miners.github.release_match import \
+    ReleaseToPullRequestMapper as OriginalReleaseToPullRequestMapper
 from athenian.api.controllers.miners.types import PRParticipants, PRParticipationKind
+from athenian.api.controllers.settings import ReleaseMatch
 from athenian.api.models.metadata.github import PullRequest
 from athenian.api.tracing import sentry_span
 
 
+class ReleaseLoader(OriginalReleaseLoader):
+    """Loader for releases."""
+
+    @classmethod
+    @sentry_span
+    async def _fetch_precomputed_releases(cls,
+                                          match_groups: Dict[ReleaseMatch, Dict[str, List[str]]],
+                                          time_from: datetime,
+                                          time_to: datetime,
+                                          account: int,
+                                          pdb: databases.Database,
+                                          index: Optional[Union[str, Sequence[str]]] = None,
+                                          ) -> pd.DataFrame:
+        cached_df = pdb.cache.dfs["releases"]
+        df = cached_df.df
+        mask = cls._match_groups_to_mask(df, match_groups)
+        releases = cached_df.filter(mask)
+        releases.sort_values("published_at", ascending=False, inplace=True)
+        releases = remove_ambigous_precomputed_releases(releases, "repository_full_name")
+        if index is not None:
+            releases.set_index(index, inplace=True)
+        else:
+            releases.reset_index(drop=True, inplace=True)
+        return releases
+
+    @classmethod
+    def _match_groups_to_mask(
+            cls,
+            df: pd.DataFrame,
+            match_groups: Dict[ReleaseMatch, Dict[str, Iterable[str]]]) -> pd.Series:
+        or_conditions, _ = match_groups_to_conditions(match_groups)
+        or_masks = [
+            (df["release_match"] == cond["release_match"]) &
+            (df["repository_full_name"].isin(cond["repository_full_name"]))
+            for cond in or_conditions
+        ]
+
+        return functools.reduce(operator.or_, or_masks)
+
+
+class ReleaseToPullRequestMapper(OriginalReleaseToPullRequestMapper):
```

Comment:
Maybe not in this PR, but I'd rather see this naming scheme:

Original -> (delete)
New classes -> prepend Preloaded

Thus

```suggestion
class PreloadedReleaseToPullRequestMapper(ReleaseToPullRequestMapper):
```